### PR TITLE
first integration with chain-validation library

### PR DESCRIPTION
**Pull request by frrist**
### Motivation
Integrate go-filecoin with the first version of the chain-validation library.

### Proposed changes
- Implement the interface required by the chain validation library.
- Add calls to the chain-validation library value transfer tests.

### Status
Ready for review

### CODE REVIEW

File: internal/pkg/validation/message.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		return nil, errors.Wrapf(err, "failed to encode abi param values: %v", params)
+	}
+	if int(method) >= len(methods) {
+		return nil, errors.Errorf("No method name for method %v", method)
+	}
+	methodName := methods[method]
+	msg := types.NewMeteredMessage(fromDec, toDec, nonce, valueDec, methodName, paramsDec, types.NewAttoFIL(gasPrice), types.NewGasUnits(uint64(gasUnit)))
+
+	return msg, nil
+	//return types.NewSignedMessage(*msg, mf.signer)
```
Comment (anorth) [inline internal/pkg/validation/message.go]: Delete

File: internal/pkg/validation/message.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+// FromSingletonAddress returns the address for a Singleton Actor.
+func (mf *MessageFactory) FromSingletonAddress(addr state.SingletonActorID) state.Address {
+	return fromSingletonAddress(addr)
+}
+
+// Maps method enumeration values to method names.
+// This will change to a mapping to method ids when method dispatch is updated to use integers.
+var methods = []types.MethodID{
+	chain.NoMethod: 0,
```
Comment (anorth) [inline internal/pkg/validation/message.go]: Go-filecoin now has method IDs. Please fill in at least a subset to demonstrate the pattern.

File: internal/pkg/validation/state.go [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+	bs := blockstore.NewBlockstore(datastore.NewMapDatastore())
+	cst := hamt.CSTFromBstore(bs)
+	treeImpl := state.NewTree(cst)
+	storageImpl := vm.NewStorageMap(bs)
+	return &StateWrapper{treeImpl, storageImpl, newKeyStore()}
+}
+
+// Cid returns the cid of the state wrappers current state
+func (s *StateWrapper) Cid() cid.Cid {
+	panic("implement me")
```
Comment (anorth) [inline internal/pkg/validation/state.go]: Blocking: I think we should replace the state.Tree with a pattern similar to that used by the Lotus adapter, storing the state root CID and hamt.CborIpldStore in the state wrapper, and then loading the state on demand when used.

File: internal/pkg/validation/state.go [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+		if err != nil {
+			return nil, nil, err
+		}
+		if err := s.Tree.SetActor(ctx, fcAddr, intAct); err != nil {
+			return nil, nil, err
+		}
+		_, err = s.Tree.Flush(ctx)
+		if err != nil {
+			return nil, nil, err
+		}
```
Comment (anorth) [inline internal/pkg/validation/state.go]: Factor out the repeated lines.

File: internal/pkg/validation/gfc_test.go [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+	factory := NewFactories()
+	suites.AccountValueTransferSuccess(t, factory, 0)
+	suites.AccountValueTransferZeroFunds(t, factory, 0)
+	suites.AccountValueTransferOverBalanceNonZero(t, factory, 0)
+	suites.AccountValueTransferOverBalanceZero(t, factory, 0)
+	suites.AccountValueTransferToSelf(t, factory, 0)
+	suites.AccountValueTransferFromKnownToUnknownAccount(t, factory, 0)
+	suites.AccountValueTransferFromUnknownToKnownAccount(t, factory, 0)
+	suites.AccountValueTransferFromUnknownToUnknownAccount(t, factory, 0)
```
Comment (anorth) [inline internal/pkg/validation/gfc_test.go]: This pattern is very fragile, and will break or miss new tests every time the suites change.
Can we instead export a list of test funcs from the validation package, and then iterate that here?

What's the `0`? Please put it in a named constant.

**Comment by frrist** [comment]
A lot has changed in the validation library wrt serialization and types. Details can be seen in https://github.com/filecoin-project/chain-validation/tree/v0.0.1. I am going to close this out and revisit later with a more concrete approach.

**Pull request closed without merging**